### PR TITLE
feat: add dark mode

### PR DESCRIPTION
**Pull request by RGHANILOO**
This pull request addresses and closes issue #114 . 

It introduces a new feature that serves a dark mode/theme based on the visitor's system-level preferences. If the visitor's system is set to dark mode, the dark theme will be applied automatically, otherwise, the default light theme will be used.

Changes Introduced
- Dark mode/theme implementation: The theme now dynamically adjusts based on the user's system preferences (using the prefers-color-scheme CSS media query).
- Default behavior: If no system preference for dark mode is detected, the website will continue to use the original light theme.
- Accessibility testing: The dark theme has been tested for accessibility using Chrome’s Lighthouse tool, with no accessibility issues reported.

Testing & Verification
Verified the dark theme using both manual testing and Chrome's Lighthouse tool.(screenshots attached for reference)
No performance or accessibility issues were found in the Lighthouse audit.

Screenshot 2024-10-02 at 00 30 43
Screenshot 2024-10-02 at 00 31 23

### CODE REVIEW

**Comment by jeremylt** [comment]
The screenshots look good to me, thanks!

There's a few spots that were hard to get the colors showing right last time I tried to define a dark mode, so I'll check a few things tomorrow.

Note, a PR just merged to main with some responsive fixes so you'll want to rebase or merge from main to resolve those conflicts. Or whatever strategy you prefer. I can help if you haven't done that before.

**Comment by RGHANILOO** [comment]
Thanks Jeremy,

I fixed the conflict issues that were showing.

I would be happy to take a look at it if you encounter any issues.

I had other suggestions relating to structure and organisation of the CSS file however I will make a separate issue relating to that.

Thanks

File: rules/style.css [hunk 1]
```diff
@@ -137,18 +137,35 @@ nav.TOC.main {
...
+    --accent-background-color: #313831;
+    --text-background-color: #ffffff;
+    --text-color: #000000;
+  }
+}
+
+@media (prefers-color-scheme: dark) {
+  :root {
+    --main-background-color: #4d4533;
+    --accent-background-color: #c6cdc6;
```
Comment (jeremylt) [inline rules/style.css:165]:
```suggestion
    --accent-background-color: #ccc5b3;
```
minor so there's more similarity between the modes

**Comment by jeremylt** [comment]
Ah, remembered where the issue was. The table headers don't have enough contrast
image

File: rules/style.css [hunk 2]
```diff
@@ -137,18 +137,35 @@ nav.TOC.main {
...
+    --main-background-color: #ccc5b3;
+    --accent-background-color: #313831;
+    --text-background-color: #ffffff;
+    --text-color: #000000;
+  }
+}
+
+@media (prefers-color-scheme: dark) {
+  :root {
+    --main-background-color: #4d4533;
```
Comment (jeremylt) [inline rules/style.css:164]:
```suggestion
    --main-background-color: #313831;
```
heck, this could work too?

**Comment by jeremylt** [comment]
Ok, see this PR: https://github.com/Eudicods/outworlds-wastes/pull/122

This will let you target the headers that have bad contrast. In dark mode we could set them to `#313831` instead of `#b2b2b2`?

**Comment by RGHANILOO** [comment]
Sure thing I will take a look at resolving it this evening.

In some of the other colour options I initially tried , I did have the issue you posted about the table heading, I will check out again and update accordingly.

Thanks

**Comment by jeremylt** [comment]
Thanks! 

For context with colors, this source also deploys a PDF and here are the colors used there: (Same as light mode)
image

File: rules/style.css (hunk 1 above)
Comment (RGHANILOO) [reply rules/style.css]: addressed in the latest commit

File: rules/style.css (hunk 2 above)
Comment (RGHANILOO) [reply rules/style.css]: addressed in the latest commit

**Comment by jeremylt** [comment]
I'll look closer in the morning but this looks great at a quick review!

**Comment by RGHANILOO** [comment]
> I'll look closer in the morning but this looks great at a quick review!

Thanks for letting me experiment and practice on a established project. 
let me know if you spot any issues.

**Comment by jeremylt** [comment]
Final double check is good. Thanks for your help!

**Comment by RGHANILOO** [comment]
> Final double check is good. Thanks for your help!

happy to help.
Thanks

**Pull request merged**